discovery-infra/download_logs.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+import os
+import json
+from glob import glob
+from argparse import ArgumentParser
+from datetime import date
+
+from logger import log
+from utils import recreate_folder
+from assisted_service_api import create_client, InventoryClient
+
+FAILED_STATUSES = ["error"]
+
+
+def main():
+    args = handle_arguments()
+    client = create_client(url=args.inventory_url)
+
+    if args.cluster_id:
+        download_logs(client, args.cluster_id, args.dest)
+    else:
+        clusters = client.clusters_list()
+
+        if not clusters:
+            log.info('No clusters were found')
+            return
+
+        for cluster in clusters:
+            if should_download_logs(cluster):
+                download_logs(client, cluster['id'], args.dest)
+
+
+def should_download_logs(cluster: dict):
+    return cluster['status'] in FAILED_STATUSES
+
+
+def download_logs(client: InventoryClient, cluster_id: str, dest: str):
+    if glob(f'{dest}/*{cluster_id}'):
+        log.info(f"Skipping. The logs of cluster {cluster_id} already exist under '{dest}' directory.")
+        return
+
+    output_folder = os.path.join(dest, f'{date.today()}_{cluster_id}')
+    recreate_folder(output_folder)
+
+    write_metadata_file(client.inventory_url, cluster_id, os.path.join(output_folder, 'metdata.json'))
+    client.download_cluster_events(cluster_id, os.path.join(output_folder, f'cluster_{cluster_id}_events.json'))
+    client.download_cluster_logs(cluster_id, os.path.join(output_folder, f'cluster_{cluster_id}_logs.tar.gz'))
+
+
+def write_metadata_file(api_url: str, cluster_id: str, file_name: str):
+    d = {'cluster_id': cluster_id,
+         'timestamp': str(date.today())}
+
+    try:
+        d['link'] = f'{get_ui_url_from_api_url(api_url)}/clusters/{cluster_id}'
+    except KeyError:
+        pass
+
+    with open(file_name, 'w') as metadata_file:
+        json.dump(d, metadata_file, indent=4)
+
+
+def get_ui_url_from_api_url(api_url: str):
+    known_urls = {
+        'https://api.openshift.com/': 'https://cloud.redhat.com/openshift/assisted-installer',
+        'https://api.stage.openshift.com/': 'https://qaprodauth.cloud.redhat.com/openshift',
+    }
+
+    for k, v in known_urls.items():
```

Comment:
just do known_urls[api_url] , u will get key error

Reply:
no need for all this code

Reply:
I thought about that. But I want to include also cases when the URL isn't same. Like 
```
https://api.stage.openshift.com/
https://api.stage.openshift.com
api.stage.openshift.com/
```

Reply:
got it